### PR TITLE
Images in cart and order history

### PR DESCRIPTION
**Pull request by MillanWang**
Checking out for 1 user will change the cart quantity for other users if they affect the seller inventory.
Changing the listing inventory to be synced to the cart items such that changing inventory can change or delete cart items. 

Updated cart page
image

Updated order history page
image

### CODE REVIEW

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/CartItemRestController.java [hunk 1]
```diff
@@ -148,4 +149,20 @@ public Map<String, Integer> deleteUserCartItem(@RequestBody Map<String, Long> pa
         map.put("inventory", inventory);
         return map;
     }
+
```
Comment (NosBos) [inline Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/CartItemRestController.java:152]: The name `updateListingInventory` as far as I can tell gives the wrong impression for what this method does. This listing inventory was already updated in the checkout method. Perhaps something like `updateAffectedCartItems` or `updateAssociatedCartItems` would make more sense

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java [hunk 2]
```diff
@@ -145,16 +146,19 @@ public Map<String, Object> putListing(@RequestBody Map<String, Object> payload)
...
     }
 
     @DeleteMapping("/listings/{id}")
     public Listing deleteListing(@PathVariable String id) {
-        Optional<Listing> listing = repo.findById(Long.parseLong(id));
+        Optional<Listing> listing = listingRepo.findById(Long.parseLong(id));
         if (listing.isPresent()) {
             listing.get().setActive(false);
-            repo.save(listing.get());
+            listing.get().setInventory(0);
```
Comment (NosBos) [inline Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java:159]: Are we setting this to 0 just to force all associated cart items to get deleted? I cant imagine we ever care about getting the previous inventory before deleting but it probably makes more sense to make a separate method in the CartItemRestController (static method similar to updateListingInventory) that just deletes all associated cart items for a given listing id

File: frontend/src/components/OrderHistory/OrderHistory.tsx [hunk 3]
```diff
@@ -27,6 +37,7 @@ export default function OrderHistory() {
     const getSellerItems = async () => {
       let response = await sendRequest(ORDER_HISTORY, "GET", {},);
       setOrderHistoryItems(response);
+      console.log(response);
```
Comment (NosBos) [inline frontend/src/components/OrderHistory/OrderHistory.tsx]: remove

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java (hunk 2 above)
Comment (MillanWang) [reply Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java:159]: Yes zeroing the inventory upon deletion is just for updating the related cart items after deletion. 
Note that there is currently nowhere in the frontend that uses this endpoint as of right now though

**Comment by NosBos** [comment]
Looks good!

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/CartItemRestController.java (hunk 1 above)
Comment (MillanWang) [reply Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/CartItemRestController.java:152]: Renamed to be more descriptive of how the listing inventory is affecting the cart quantity

File: frontend/src/components/OrderHistory/OrderHistory.tsx (hunk 3 above)
Comment (MillanWang) [reply frontend/src/components/OrderHistory/OrderHistory.tsx]: Removed

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java (hunk 2 above)
Comment (MillanWang) [reply Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java:159]: I feel that the extensibility in the newly renamed `updateAssociatedCartItems_listingInventoryUpdated` static method would be sufficient to handle the only deletion case as that use case is partially covered by zeroing an active listing's inventory already and I'd prefer that the code not be repeated

**Review by NosBos — APPROVED** [review]

**Pull request merged**